front/components/assistant/conversation/ConversationMenu.tsx

```diff
@@ -0,0 +1,202 @@
+import {
+  useConversationMenu,
+  useDeleteConversation,
+} from "@app/lib/swr/conversations";
+import { useUser } from "@app/lib/swr/user";
+import type { ConversationWithoutContentType, WorkspaceType } from "@app/types";
+import { asDisplayName } from "@app/types/shared/utils/string_utils";
+
+export function ConversationMenu({
+  activeConversationId,
+  conversation,
+  baseUrl,
+  owner,
+}: {
+  activeConversationId: string | null;
+  conversation: ConversationWithoutContentType | null;
+  baseUrl: string;
+  owner: WorkspaceType;
+}) {
+  const { user } = useUser();
+  const router = useRouter();
+  const sendNotification = useSendNotification();
+  const { joinConversation, action, users, agents } = useConversationMenu({
+    ownerId: owner.sId,
+    conversationId: activeConversationId,
+    userId: user?.sId || null,
+  });
+  const [showDeleteDialog, setShowDeleteDialog] = useState<boolean>(false);
+  const [showLeaveDialog, setShowLeaveDialog] = useState<boolean>(false);
+  const [showRenameDialog, setShowRenameDialog] = useState<boolean>(false);
+
+  const shareLink = `${baseUrl}/w/${owner.sId}/assistant/${activeConversationId}`;
+
+  const doDelete = useDeleteConversation(owner);
+  const leaveOrDelete = useCallback(async () => {
+    const res = await doDelete(conversation);
+    if (res) {
+      void router.push(`/w/${owner.sId}/assistant/new`);
+    }
+  }, [conversation, doDelete, owner.sId, router]);
+
+  const copyConversationLink = useCallback(async () => {
+    await navigator.clipboard.writeText(shareLink || "");
+    sendNotification({ type: "success", title: "Link copied !" });
+  }, [shareLink, sendNotification]);
+
+  if (!activeConversationId) {
+    return null;
+  }
+
+  const ConversationActionMenuItem = () => {
+    switch (action) {
+      case "delete":
+        return (
+          <DropdownMenuItem
+            label="Delete"
+            onClick={() => setShowDeleteDialog(true)}
+            icon={TrashIcon}
+          />
+        );
+      case "leave":
+        return (
+          <DropdownMenuItem
+            label="Leave"
+            onClick={() => setShowLeaveDialog(true)}
+            icon={XMarkIcon}
+          />
+        );
+      case "join":
+        return (
+          <DropdownMenuItem
+            label="Join"
+            onClick={joinConversation}
+            icon={PlusCircleIcon}
+          />
+        );
+    }
+  };
+
+  return (
+    <>
+      <DeleteConversationsDialog
+        isOpen={showDeleteDialog}
+        type="selection"
+        selectedCount={1}
+        onClose={() => setShowDeleteDialog(false)}
+        onDelete={() => {
+          setShowDeleteDialog(false);
+          void leaveOrDelete();
+        }}
+      />
+      <LeaveConversationDialog
+        isOpen={showLeaveDialog}
+        onClose={() => setShowLeaveDialog(false)}
+        onLeave={() => {
+          setShowLeaveDialog(false);
+          void leaveOrDelete();
+        }}
+      />
```

Comment:
I really feel like those two are doing the same thing

Reply:
As seen together : DeleteConversationDialog is already handling too many things:
- type "all" or "selected" as props
- different action depending on the type
- creating a lot of overhead in SideBarMenu

Conclusion : the actions should be scoped to the Dialog for more readability --> new refactor PR inc. with LeaveConversationDialog and DeleteConversationDialog